### ISSUE TITLE
[v5] react-query hooks ignore the `Data` generic passed to them

### ISSUE THREAD

**Issue by Eddie-CooRo** [anymaniax/orval#158]
The react-query hooks accept two generics. one for the Data and another one for the Error.
Which are supposed to give the user the flexibility to override the generated types on the fly.
But it's not working as expected.
## What are the steps to reproduce this issue?
1. Call a hook and pass `any` as the `Data` generic
```ts
const { data } = useSampleQuery<any, any>();
```

## What happens?
The type of the `data` variable doesn't change. 

## What were you expecting to happen?
The type of `data` should be `any`.

## Any other comments?
Here is the reason this is happening.
Consider the following generated hook.
```ts
export const useShowPetById = <
  Data extends unknown = unknown,
  Error extends unknown = unknown
>(
  petId: string,
  version = 1,
  options?: {
    query?: UseQueryOptions<AsyncReturnType<typeof showPetById>, Error>;
    request?: SecondParameter<typeof customInstance>;
  },
) => {
  const { query: queryOptions, request: requestOptions } = options || {};

  const queryKey =
    queryOptions?.queryKey ?? getShowPetByIdQueryKey(petId, version);
  const query = useQuery<AsyncReturnType<typeof showPetById>, Error>(
    queryKey,
    () => showPetById<Data>(petId, version, requestOptions),
    { enabled: !!(version && petId), ...queryOptions },
  );

  return {
    queryKey,
    ...query,
  };
};

```
The `Data` generic is only used in one place. when calling the fetcher, `showPetById`. It's passed as a generic to it, but it's completely useless as the `TData` type of the whole `useQuery` is manually set to `AsyncReturnType<typeof showPetById>` (without `Data` passed by user) So the `Data` type provided by the user is ignored altogether.

### Solution
There is no way to apply the `Data` to the type of the `showPetById` with the current approach.  
As the function generics are bound with the function call, one workaround might be this:
```diff
export const useShowPetById = <
  Data extends unknown = unknown,
  Error extends unknown = unknown
>(
  petId: string,
  version = 1,
  options?: {
    query?: UseQueryOptions<AsyncReturnType<typeof showPetById>, Error>;
    request?: SecondParameter<typeof customInstance>;
  },
) => {
  const { query: queryOptions, request: requestOptions } = options || {};

  const queryKey =
    queryOptions?.queryKey ?? getShowPetByIdQueryKey(petId, version);
+ const fetcher = () => showPetById<Data>(petId, version, requestOptions);
- const query = useQuery<AsyncReturnType<typeof showPetById>, Error>(
+ const query = useQuery<AsyncReturnType<typeof fetcher>, Error>(
   queryKey,
-  () => showPetById<Data>(petId, version, requestOptions),
+  fetcher,
   { enabled: !!(version && petId), ...queryOptions },
  );

  return {
    queryKey,
    ...query,
  };
};
```

## What versions are you using?
**Package Version:** next

**Comment by anymaniax** [comment]
should be solve with `5.4.7`

**Comment by OliverDudgeon** [comment]
After this fix,
```ts
const mutation = useSomeMutation();

const res = mutation.mutateAsync({ ... })
```

always gives a `unknown` value for `res`.

**Comment by anymaniax** [comment]
sorry about that I will do a fix today normally

**Comment by anymaniax** [comment]
fixed with `5.4.8`
